sqlite/vdbe.c

```diff
@@ -2495,7 +2495,7 @@ case OP_Ge: {             /* same as TK_GE, jump, in1, in3 */
     pOut = &aMem[pOp->p2];
     iCompare = res;
     res2 = res2!=0;  /* For this path res2 must be exactly 0 or 1 */
-    if( (pOp->p5 & SQLITE_KEEPNULL)!=0 ){
+    if( (pOp->p5 & SQLITE_KEEPNULL)==SQLITE_KEEPNULL ){
```

Comment:
STOREP2 is 0x20 -- `0010 0000`
KEEPNULL is 0x30 -- `0011 0000`
If we reach the code at this line, it is guaranteed that STOREP2 is set, i.e. p5 is `**1* ****`.
But we want to check if KEEPNULL is also set here, and `**1* **** & 0011 0000` is always nonzero. So instead we want to check `**1* **** & 0011 0000 == 0011 0000` to make sure both bits are set.